### DISCUSSION TITLE
Does Valtio support true private class fields (ES2022)?

### DISCUSSION THREAD

**Discussion by pastelmind** [pmndrs/valtio#746] [Q&A] [answered]
I heard [`Proxy`-based state management doesn't play well with true private fields](https://github.com/tc39/proposal-private-fields/issues/102#issuecomment-397115889) (standardized in ES2022) in general. Does Valtio support them? Should I avoid using `#`-prefixed fields and methods in general?

```js
class Person {
  #age = 0

  getAge() {
    return this.#age
  }

  updateAge(value) {
    this.#age = value
  }
}

const p = proxy(new Person())
p.updateAge(100) // Will this correctly notify listeners that use p.getAge()?
```

Also, what about Symbol keys? They can be used as 'soft' private fields. Does Valtio support them, too?

```js
const key = Symbol('Person.age')

class Person {
  [key] = 0

  getAge() {
    return this[key]
  }

  updateAge(value) {
    this[key] = value
  }
}

const p = proxy(new Person())
p.updateAge(100) // Will this correctly notify listeners that use p.getAge()?
```

**Comment by dai-shi** [comment, marked as answer]
I haven't tried, but I don't think private fields work with proxies.
Symbol keys should work fine.
